src/Compilers/CSharp/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.cs

```diff
@@ -71,6 +85,43 @@ private DynamicAnalysisInjector(MethodSymbol method, BoundStatement methodBody,
             }
         }
 
+        private static bool IsExcludedFromCodeCoverage(MethodSymbol method)
+        {
+            if (method.ContainingType is SourceNamedTypeSymbol sourceNamedType &&
```

Comment:
>if (method.ContainingType is SourceNamedTypeSymbol sourceNamedType && [](start = 12, length = 69)

It feels like it would be better to not rely on specific implementation types in this function. I would simply do GetAttributes and check them for the ones we are interested in.